src/Database/Expression/WindowInterface.php

```diff
@@ -80,69 +48,77 @@ public function partition($partitions);
     public function order($fields);
 
     /**
-     * Adds a range frame clause to the window. Only one frame clause can be
-     * specified per window.
+     * Adds a simple range frame to the window.
      *
-     * `$start` assumes `PRECEDING`, and `$end` assumes `FOLLOWING`. Both can be
-     * overriden by passing an array with the order as the key. The SQL standard
-     * for ordering must be followed.
+     * `$start`:
+     *  - `0` - 'CURRENT ROW'
+     *  - `null` - 'UNBOUNDED PRECEDING'
+     *  - offset - 'offset PRECEDING'
      *
-     * ```
-     * // this is produces 'ROWS BETWEEN 1 PRECEDING AND 2 FOLLOWING'
-     * $window->rows(1, 2);
+     * `$end`:
+     *  - `0` - 'CURRENT ROW'
+     *  - `null` - 'UNBOUNDED FOLLOWING'
+     *  - offset - 'offset FOLLOWING'
      *
-     * // this is the same as 'ROWS 1 FOLLOWING`
-     * $window->rows([WindowInterface::FOLLOWING => 1]);
-     * ```
+     * If you need to use 'FOLLOWING' with frame start or
+     * 'PRECEDING' with frame end, use `frame()` instead.
      *
-     * You can use `null` for 'UNBOUNDED' and `0` for 'CURRENT ROW'.
-     *
-     * ```
-     * // this is produces 'ROWS CURRENT ROW'
-     * $window->rows(0);
+     * @param int|null $start Frame start
+     * @param int|null $end Frame end
+     *  If not passed in, only frame start SQL will be generated.
+     * @return $this
+     */
+    public function range(?int $start, ?int $end = 0);
+
+    /**
+     * Adds a simple rows frame to the window.
      *
-     * // this is produces 'ROWS UNBOUNDED PRECEDING'
-     * $window->rows(null)
-     * ```
+     * See `range()` for details.
      *
-     * @param array|int|null $start Frame start
-     * @param array|int|null $end Frame end
+     * @param int|null $start Frame start
+     * @param int|null $end Frame end
      *  If not passed in, only frame start SQL will be generated.
      * @return $this
      */
-    public function range($start, $end = 0);
+    public function rows(?int $start, ?int $end = 0);
 
     /**
-     * Adds a rows frame clause to the window. Only one frame clause can be
-     * specified per window.
+     * Adds a simple groups frame to the window.
      *
-     * See `range()` for details on `$start` and `$end` format.
+     * See `range()` for details.
      *
-     * @param array|int|null $start Frame start
-     * @param array|int|null $end Frame end
+     * @param int|null $start Frame start
+     * @param int|null $end Frame end
      *  If not passed in, only frame start SQL will be generated.
      * @return $this
      */
-    public function rows($start, $end = 0);
+    public function groups(?int $start, ?int $end = 0);
```

Comment:
These changes are great :clap: The simpler types are much better, and the additional methods to handle the special cases will be simper to read down the road.